src/GoTrueClient.ts

```diff
@@ -92,19 +92,35 @@ export default class GoTrueClient {
   async signUp({
     email,
     password,
-  }: UserCredentials): Promise<{ data: Session | null; user: User | null; error: Error | null }> {
+  }: UserCredentials): Promise<{ data: Session | User | null; user: User | null; error: Error | null }> {
     try {
       this._removeSession()
 
       const { data, error } = await this.api.signUpWithEmail(email!, password!)
-      if (error) throw error
 
-      if (data?.user?.confirmed_at) {
-        this._saveSession(data)
+      if (error) {
+        throw error
+      }
+
+      if (!data) {
+        throw 'An error occurred on sign up.'
+      }
+
+      let session: Session
+      let user: User | null = null
+
+      if ((data as Session).access_token) {
```

Comment:
We need to add checks here to determine if GoTrue has returned a session or a user.

If "autoconfirm" is on, it will return a full session